core/translate/expr.rs

```diff
@@ -1552,6 +1552,112 @@ pub fn translate_expr(
+                    MathFunc::Acos
+                    | MathFunc::Acosh
+                    | MathFunc::Asin
+                    | MathFunc::Asinh
+                    | MathFunc::Atan
+                    | MathFunc::Atanh
+                    | MathFunc::Ceil
+                    | MathFunc::Ceiling
+                    | MathFunc::Cos
+                    | MathFunc::Cosh
+                    | MathFunc::Degrees
+                    | MathFunc::Exp
+                    | MathFunc::Floor
+                    | MathFunc::Ln
+                    | MathFunc::Log10
+                    | MathFunc::Log2
+                    | MathFunc::Radians
+                    | MathFunc::Sin
+                    | MathFunc::Sinh
+                    | MathFunc::Sqrt
+                    | MathFunc::Tan
+                    | MathFunc::Tanh
+                    | MathFunc::Trunc => {
+                        let args = if let Some(args) = args {
+                            if args.len() != 1 {
+                                crate::bail_parse_error!(
+                                    "{} function with not exactly 1 argument",
+                                    math_func
+                                );
+                            }
+                            args
+                        } else {
+                            crate::bail_parse_error!("{} function with no arguments", math_func);
+                        };
+
+                        let regs = program.alloc_register();
+                        translate_expr(
+                            program,
+                            referenced_tables,
+                            &args[0],
+                            regs,
+                            precomputed_exprs_to_registers,
+                        )?;
+                        program.emit_insn(Insn::Function {
+                            constant_mask: 0,
+                            start_reg: regs,
+                            dest: target_register,
+                            func: func_ctx,
+                        });
+                        Ok(target_register)
+                    }
+
+                    // Binary math functions
+                    MathFunc::Atan2 | MathFunc::Mod | MathFunc::Pow | MathFunc::Power => {
+                        let args = if let Some(args) = args {
+                            if args.len() != 2 {
+                                crate::bail_parse_error!(
+                                    "{} function with not exactly 2 argument",
+                                    math_func
+                                );
+                            }
+                            args
+                        } else {
+                            crate::bail_parse_error!("{} function with no arguments", math_func);
+                        };
+                        for arg in args.iter() {
+                            let reg = program.alloc_register();
```

Comment:
you might get in trouble with this if the arguments are expressions that themselves allocate registers, since the final `Insn::Function` takes a `start_reg` and assumes that all of the arguments are in contiguous registers.

so you probably want to allocate the registers upfront instead of in the loop.

can you add a tcl test (even if temporary) that verifies that something like `pow(users.id+2, 2)` actually works?